cirq/contrib/quirk/url_to_circuit.py

```diff
@@ -193,36 +271,52 @@ def msg(error):
                 if cell is not None:
                     c[i] = modifier(cell)
 
-    # Extract circuit operations from modified cells.
-    result = circuits.Circuit()
-    for col in parsed_cols:
-        basis_change = circuits.Circuit(
-            cell.basis_change() for cell in col if cell is not None)
-        body = circuits.Circuit(
-            cell.operations() for cell in col if cell is not None)
-        result += basis_change
-        result += body
-        result += basis_change**-1
+    gate_count = sum(0 if cell is None else cell.gate_count()
+                     for col in parsed_cols
+                     for cell in col)
+
```

Comment:
Can we raise `ValueError` for billion laughs here instead of l.208? Then we never have to instantiate a `CompositeCell` with dangerous cell structure and don't need the extra complexity to ensure lazy traversals.

Reply:
what's `l.208`? Later?

I tried doing this, and the runtime of the test to detect the attack increased from a fraction of a section to tens of seconds. The problem is that this approach performs O(MAX_OP_COUNT) work instead of O(INPUT LENGTH) work when defending against an attack, because the largest composite gate under the limit is still built.